File: April-Challenges-2021/Day-21/Brick-Wall.go
func leastBricks(wall [][]int) int {
	ret := 0
	hashTable := map[int]int{}
	
	for i := range wall {
		curWidth := 0
		for j := 0; j < len(wall[i])-1; j++ {
			curWidth += wall[i][j]
			hashTable[curWidth]++
		}
	}
	for _, v := range hashTable {
		if v > ret {
			ret = v
		}
	}
	return len(wall) - ret  
}

File: April-Challenges-2021/Day-21/Brick-Wall.py
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        dp = {}
        maxCount = 0
        
        for row in wall:
            idx = 0
            maxIdx = sum(row)
            for brick in row:
                idx += brick
                if idx == maxIdx:
                    continue
                count = dp.get(idx, 0) + 1
                dp[idx] = count
                if count > maxCount:
                    maxCount = count
            
        return len(wall) - maxCount
